tee-worker/enclave-runtime/src/rpc/worker_api_direct.rs

```diff
@@ -103,6 +106,32 @@ where
 		Ok(json!(json_value.to_hex()))
 	});
 
+	// author_getEnclaveId
+	let rsa_pubkey_name: &str = "author_getEnclaveId";
+	io.add_sync_method(rsa_pubkey_name, move |_: Params| {
+		let shielding_key_repository = match GLOBAL_SHIELDING_KEY_REPOSITORY_COMPONENT.get() {
+			Ok(s) => s,
+			Err(e) => {
+				let error_msg: String = format!("{:?}", e);
+				debug!("{:?}", e);
+				return Ok(json!(compute_hex_encoded_return_error(error_msg.as_str())))
+			},
+		};
+		let vc_pubkey = shielding_key_repository
+			.retrieve_key()
+			.and_then(|keypair| keypair.derive_ed25519().map(|keypair| keypair.public().to_hex()))
+			.ok();
```

Comment:
`ok()` leaves an `Option` behind, that's probably why you need `slice(1)`  below.

We should unwrap it's value - either a direct `unwrap()` or handle the error if you want

Reply:
I thought we wanted to avoid `unwrap` (and panics in general)?

Reply:
I corrected myself during a call with Zhouran earlier but forgot to edit the comment, sorry!

Indeed we should avoid panics in enclave, so errors should be handled here.